R/MsBackendMetaboLights.R

```diff
@@ -0,0 +1,267 @@
+#'
+#' @importMethodsFrom ProtGenerics dataOrigin
+#'
+#' @importFrom methods callNextMethod
+#'
+#' @exportMethod backendInitialize
+setMethod(
+    "backendInitialize", "MsBackendMetaboLights",
+    function(object, mtblsId = character(), assayId = character(),
+             filePattern = "mzML$|CDF$|cdf$|mzXML$", offline = FALSE, ...) {
+        dots <- list(...)
+        if (any(names(dots) == "data"))
+            stop("Parameter 'data' is not supported for ",
+                 "'MsBackendMetaboLights'. A 'MsBackendMetaboLights' object ",
+                 "can only be instantiated using 'backendInitialize()'.")
+        if (length(mtblsId) != 1)
+            stop("Parameter 'mtblsId' is required and can only be a single ID ",
+                 "of a MetaboLights data set.")
+        if (offline)
+            mdata <- .mtbls_data_files_offline(mtblsId, assayId, filePattern)
+        else mdata <- .mtbls_data_files(mtblsId, assayId, filePattern)
+        object <- callNextMethod(object, files = mdata$rpath)
+        object@mtblsId <- mtblsId
+        object@assays <- unique(mdata$mtbls_assay_name)
+        idx <- match(dataOrigin(object),
+                     normalizePath(mdata$rpath, mustWork = FALSE))
+        object@spectraData$mtbls_id <- mdata$mtbls_id[idx]
+        object@spectraData$mtbls_assay_name <- mdata$mtbls_assay_name[idx]
+        object@spectraData$derived_spectral_data_file <-
+            mdata$derived_spectral_data_file[idx]
+        object
+    })
+
+#' @rdname MsBackendMetaboLights
+#'
+#' @importFrom ProtGenerics backendMerge
+#'
+#' @exportMethod backendMerge
+setMethod(
+    "backendMerge", "MsBackendMetaboLights",
+    function(object, ...) {
+        stop("Merging of backends of type 'MsBackendMetaboLights' is not ",
+             "supported. Use 'setBackend()' to change to a backend that ",
+             "supports merging, such as the 'MsBackendMemory'.")
+    })
+
+
+################################################################################
+##
+## File caching utils
+##
+################################################################################
+
+#' Get information on data files for a given MTBLS ID/assay eventually
+#' downloading and caching them. This function needs an active internet
+#' connection as it queries the MTBLS ftp server for available data files
+#' that are then cached. The function returns the **local** file names
+#' **from the cache**.
+#'
+#' The function:
+#' - retrieves all "Derived Data Files" for all assays (or for specified assays)
+#'   for one MetaboLights ID.
+#' - uses BiocFileCache to cache these files, i.e. downloading them if they
+#'   are not yet cached.
+#' - returns a `data.frame` with all information.
+#'
+#' This `data.frame` has one row per data file with columns:
+#' - `"rid"`: the BiocFileCache ID of each file.
+#' - `"mtbls_id"`: the MTBLS ID
+#' - `"mtbls_assay_name"`: the name of the assay file for each data file
```

Comment:
maybe `assayId` shoud be renamed to `assayName` ? especially if we expect character() input ?

Reply:
yes, indeed.

Reply:
I will change that.